cmd/status/cmdstatus.go

```diff
@@ -0,0 +1,214 @@
+
+	period    time.Duration
+	pollUntil string
+	timeout   time.Duration
+	output    string
+
+	inventoryClientFactoryFunc func(cmdutil.Factory) (inventory.InventoryClient, error)
+	pollerFactoryFunc          func(cmdutil.Factory) (poller.Poller, error)
+}
+
+// runE implements the logic of the command and will delegate to the
+// poller to compute status for each of the resources. One of the printer
+// implementations takes care of printing the output.
+func (r *StatusRunner) runE(cmd *cobra.Command, args []string) error {
+	_, err := common.DemandOneDirectory(args)
+	if err != nil {
+		return err
+	}
+
+	var reader manifestreader.ManifestReader
+	readerOptions := manifestreader.ReaderOptions{
+		Factory:   r.factory,
+		Namespace: metav1.NamespaceDefault,
+	}
+	if len(args) == 0 {
+		reader = &manifestreader.StreamManifestReader{
+			ReaderName:    "stdin",
+			Reader:        cmd.InOrStdin(),
+			ReaderOptions: readerOptions,
+		}
+	} else {
+		reader = &manifestreader.PathManifestReader{
+			Path:          args[0],
+			ReaderOptions: readerOptions,
+		}
+	}
+	infos, err := reader.Read()
+	if err != nil {
+		return err
+	}
+
+	// Find the inventory template among the manifests.
+	inv, _, err := inventory.SplitInfos(infos)
+	if err != nil {
+		return err
+	}
+
+	invClient, err := r.inventoryClientFactoryFunc(r.factory)
+	if err != nil {
+		return err
+	}
+
+	// Based on the inventory template manifest we look up the inventory
+	// from the live state using the inventory client.
+	identifiers, err := invClient.GetClusterObjs(inv)
+	if err != nil {
+		return err
+	}
+
+	// Exit here if the inventory is empty.
+	if len(identifiers) == 0 {
+		_, _ = fmt.Fprint(cmd.OutOrStdout(), "no resources found in the inventory\n")
+		return nil
+	}
+
+	statusPoller, err := r.pollerFactoryFunc(r.factory)
+	if err != nil {
+		return err
+	}
+
+	// Fetch a printer implementation based on the desired output format as
+	// specified in the output flag.
+	printer, err := printers.CreatePrinter(r.output, genericclioptions.IOStreams{
+		In:     cmd.InOrStdin(),
+		Out:    cmd.OutOrStdout(),
+		ErrOut: cmd.ErrOrStderr(),
+	})
+	if err != nil {
+		return errors.WrapPrefix(err, "error creating printer", 1)
+	}
+
+	// If the user has specified a timeout, we create a context with timeout,
+	// otherwise we create a context with cancel.
+	ctx := context.Background()
+	var cancel func()
+	if r.timeout != 0 {
+		ctx, cancel = context.WithTimeout(ctx, r.timeout)
+	} else {
+		ctx, cancel = context.WithCancel(ctx)
+	}
+	defer cancel()
+
+	// Choose the appropriate ObserverFunc based on the criteria for when
+	// the command should exit.
+	var cancelFunc collector.ObserverFunc
+	switch r.pollUntil {
+	case "known":
+		cancelFunc = getAllKnownNotifierFunc(cancel)
+	case "current":
+		cancelFunc = getDesiredStatusNotifierFunc(cancel, status.CurrentStatus)
+	case "deleted":
+		cancelFunc = getDesiredStatusNotifierFunc(cancel, status.NotFoundStatus)
+	case "forever":
```

Comment:
This option waits forever for the status of resources ? If so, what if there is bad network or it's taking too long. Should the command block forever?

Reply:
This is just an option for users who just want to monitor the state of the resources without a clear end. They can shut it down by interrupting the program with ctrl+c (or equivalent).

Reply:
okay